components/shared/shell/Header.tsx

```diff
@@ -0,0 +1,106 @@
+import Link from 'next/link';
+import React from 'react';
+import { useSession } from 'next-auth/react';
+import {
+  ArrowRightOnRectangleIcon,
+  Bars3Icon,
+  SunIcon,
+  UserCircleIcon,
+} from '@heroicons/react/24/outline';
+import { ChevronDownIcon } from '@heroicons/react/20/solid';
+import useTheme from 'hooks/useTheme';
+import env from '@/lib/env';
+import { signOut } from 'next-auth/react';
+
+interface HeaderProps {
+  setSidebarOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const Header = ({ setSidebarOpen }: HeaderProps) => {
+  const { toggleTheme } = useTheme();
+  const { status, data } = useSession();
+
+  if (status === 'loading' || !data) {
+    return null;
+  }
+
+  const user = data.user;
+
+  return (
+    <div className="sticky top-0 z-40 flex h-14 shrink-0 items-center border-b px-4 sm:gap-x-6 sm:px-6 lg:px-8 bg-white">
+      <button
+        type="button"
+        className="-m-2.5 p-2.5 text-gray-700 lg:hidden"
+        onClick={() => setSidebarOpen(true)}
+      >
+        <span className="sr-only">Open sidebar</span>
+        <Bars3Icon className="h-6 w-6" aria-hidden="true" />
+      </button>
+      <div className="flex flex-1 gap-x-4 self-stretch lg:gap-x-6">
+        <div className="relative flex flex-1"></div>
+        <div className="flex items-center gap-x-4 lg:gap-x-6">
+          <div className="dropdown dropdown-end dropdown-hover">
+            <div className="flex items-center cursor-pointer" tabIndex={0}>
+              <span className="hidden lg:flex lg:items-center">
+                <button
```

Comment:
The dropdown activation does not work on 'Space' or 'Enter' key press on button.